lib/callableFunction.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var _ = require('lodash');
+var request = require('request');
+
+var _isDatabaseFunc = function(eventTrigger) {
+  return /firebase.database/.test(eventTrigger.eventType);
+};
+
+var _substituteParams = function(resource, params) {
+  var wildcards = resource.match(new RegExp('{[^/{}]*}', 'g'));
+  _.forEach(wildcards, function(wildcard) {
+    var wildcardNoBraces = wildcard.slice(1, -1); // .slice removes '{' and '}' from wildcard
+    var sub = _.get(params, wildcardNoBraces);
+    if (sub) {
+      resource = _.replace(resource, wildcard, sub);
+    } else {
+      resource = _.replace(resource, wildcard, wildcardNoBraces + '1');
+    }
+  });
+  return resource;
+};
+
+var _call = function(data, opts) {
+  opts = opts || {};
+  if (this.httpsTrigger) {
+    this.controller.call(this.name, data || {});
+  } else if (this.eventTrigger) {
+    if (_isDatabaseFunc(this.eventTrigger)) {
+      var operationType = _.last(this.eventTrigger.eventType.split('.'));
+      var dataPayload;
+      if (operationType === 'create') {
+        dataPayload = {
+          data: null,
+          delta: data
+        };
+      } else if (operationType === 'update' || operationType === 'write') {
+        dataPayload = {
+          data: data.before,
+          delta: data.after
```

Comment:
This is actually a likely breaking change coming to the SDK ("before"/"after" instead of delta)